File: data/EXP01/golang_no_system_var/2348/2348.output_3.go
func zeroFilledSubarray(nums []int) int {
    count := 0
    prefixSum := make(map[int]int)
    prefixSum[0] = 1
    sum := 0
    
    for _, num := range nums {
        if num == 0 {
            sum += 1
        } else {
            sum -= 1
        }
        
        if freq, exist := prefixSum[sum]; exist {
            count += freq
        }
        prefixSum[sum] += 1
    }
    
    return count
}

File: data/EXP01/python_3/2348/2348.output_3.py
# @lc app=leetcode id=2348 lang=python
class Solution:
    def zeroFilledSubarray(self, nums):
        count = 0
        for i in range(len(nums)):
            if nums[i] == 0:
                count += 1
                j = i - 1
                while j >= 0 and nums[j] == 0:
                    count += 1
                    j -= 1
        return count